### DISCUSSION TITLE
How to handle errors in different env?

### DISCUSSION THREAD

**Discussion by matthiasPOE** [rectorphp/rector#8642] [Q&A] [answered]
Hi everyone,

bit stumped at the moment. Yesterday my github actions pipeline ran through with no errors. I merged my changes into dev and all of a sudden it failed. After a bit of back and forth I decided to rerun the old run from yesterday and it suddenly fails too on the same commit that was green at that point.

![Screenshot 2024-05-16 at 15 29 53](https://github.com/phpstan/phpstan/assets/162434935/07d9ddc6-5edb-44b7-902b-e81c372d7257)

A bit about the setup: I build php docker images which I run locally & also build in actions and reuse them in github actions so it should be fairly similar. I did think this might me a mac issue locally so I booted up a codespace and rector runs through fine on it too. It only gets stuck in github actions now with 

```
Applied rules:
 * RemoveParentCallWithoutParentRector
```

I am unable to reproduce the same diff locally or in codespace using the same docker image. Unfortunately I cant really share any code but im wondering does anyone have any experience on how to best debug issues that you cant reproduce locally? I've also tried going up and down rector & phpstan versions with no luck.

currently on:

```
    "phpstan/phpstan": "1.10.67",
    "rector/rector": "1.0.5",
```

My composer run script:
```
    "rector": [
      "Composer\\Config::disableProcessTimeout",
      "rector --dry-run --clear-cache"
    ],
```

I dont want to simply ignore this rule because I need to try and reproduce it locally otherwise I cant trust local / pipeline results.

**Comment by samsonasik** [comment]
Could you try latest rector/rector:dev-main?

```diff
-    "rector/rector": "1.0.5",
+    "rector/rector": "dev-main",
```

and verify if using dev-main fix the issue?

**Reply by matthiasPOE** [reply]
Hi @samsonasik thanks for the response, unfortunately not:

```
Applied rules:
 * RemoveParentCallWithoutParentRector
 [OK] 1 file would have changed (dry-run) by Rector   
```

Weirdly enough I can only get it to reproduce locally if I remove the `extends ParentClassName` from the class itself that errors so its like as if in the github actions pipeline it doesnt see the parent class but locally + codespace it sees it. Ive also tried downsizing the codespace to 2 cores but it still runs fine there too.

**Reply by samsonasik** [reply]
I am not sure we can help if there is no code/config to reproduce it, if you cant' share the code/config for us, and don't use the rule, you can just skip the rule :) 

https://getrector.com/documentation/ignoring-rules-or-paths

**Reply by matthiasPOE** [reply]
Yea I will try and get a reproducible simple example to share thank you. 

Unfortunately I would like to use the rule but more importantly I would like to reproduce it locally simply to be able to trust the process. What happens if the next rule does the same? I cant just disable all the rules because they produce diffs in different envs (with same php version).

**Reply by samsonasik** [reply, marked as answer]
That probably due to autoload issue, for example, on my CodeIgniter4 module, I need to provide custom bootstrap to be loaded from rector config, by using class_exists() check 

https://github.com/samsonasik/ci4-album/blob/1a9821cc086013c88ced072c3815acd3b0844b94/bootstrap.php#L17

**Reply by matthiasPOE** [reply]
You are correct!!!!! Thank you.

I was under the wrong assumption that due to it showing the files being processed:

 700/700 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

I had assumed it can see them & that it was loading them. However the class exists check didnt return anything until I included more bootstrap files.

EDIT: Now to find out why it works locally & codespace but not in the pipeline
